model/src/main/kotlin/Scope.kt

```diff
@@ -90,4 +91,42 @@ data class Scope(
 
         return getTreeDepthRec(dependencies)
     }
+
+    /**
+     * Return the shortest path to each dependency in this scope. The path to a dependency is defined by the nodes of
+     * the dependency tree that need to be passed to get to the dependency. For direct dependencies the shortest path is
+     * empty.
+     */
+    @JsonIgnore
+    fun getShortestPaths(): Map<Identifier, List<Identifier>> {
```

Comment:
Did you consider representing the shortest path tree by `Map<Identifier, Identifier>` which maps each `id` to its parents? That'd completely remove the redundancy so the result could not be inconsistent and it'd be more memory friendly. Execution time wise that'd speed-up things in case an analog adjustment to the implementation is made to not manage lists of parents anymore: Note the copying of the lists.

Reply:
That again would require that subtrees are always identical. In the example above with this you could get the wrong path with this approach: `C:1.1->B:1.0->A:1.0`